marker_rustc_driver/src/conversion/marker/pat.rs

```diff
@@ -41,7 +38,7 @@ impl<'ast, 'tcx> MarkerConverterInner<'ast, 'tcx> {
                 let lhs = lhs_map.get(id);
                 #[allow(clippy::unnecessary_unwrap, reason = "if let sadly breaks rustfmt")]
                 if pat.is_none() && matches!(mutab, rustc_ast::Mutability::Not) && lhs.is_some() {
-                    PatKind::Place(*lhs.unwrap(), CtorBlocker::new())
```

Comment:
Could we get rid `CtorBlocker` type in this PR? That seems like an uncommon hack

Reply:
It is unusual, but I sadly don't see a better option. If we take [`LitExprKind::UnaryOp`] for example. We want to make sure that the variant is only filled with a `-` operator, when it's used as a sign on a literal. Otherwise, it would not be a `LitExprKind`. Without the blocker, we would always need to make sure that the kind wasn't constructed by the user and nests some unexpected nodes.

Ideally, I would have liked to use `#[non_exhaustive]` but that only works on variants with fields. Tuple variants are made private, with that attribute. We could use variants with named fields, but I generally like tuple variants more for matching. I find them more readable and less verbose.

I haven't found or seen a better hack yet, and this one seems to work pretty well IMO, even if it is a hack